blockchain/blockchain.go

```diff
@@ -1089,7 +1088,7 @@ func (bc *BlockChain) Stop() {
 				}
 
 				logger.Info("Writing cached state to disk", "block", recent.Number(), "hash", recent.Hash(), "root", recent.Root())
-				if err := triedb.Commit(recent.Root(), true, statedb.NoDataArchivingPreparation); err != nil {
+				if err := triedb.Commit(recent.Root(), true, number-offset); err != nil {
```

Comment:
```suggestion
				if err := triedb.Commit(recent.Root(), true, recent.Number()); err != nil {
```
I think it looks better.